crates/viewer/re_viewer/src/history.rs

```diff
@@ -277,6 +273,11 @@ impl private::Sealed for History {}
 
 impl HistoryExt for History {
     fn push_entry(&self, entry: HistoryEntry) -> Result<(), JsValue> {
+        // Check if this is the exact same entry as before, if so don't do anything.
+        if self.current_entry()?.unwrap_or_default() == entry {
```

Comment:
We have a handler for `popstate` which does non-idempotent things. Are we sure this is the right behavior? In what case can you end up with duplicate entries, and why would that be a problem requiring a solution here?

Reply:
this would happen for instance if a recording gets activated that's already active. Maybe this should be a separate method for "push only if not new" of sorts?
I feel like I'm not understanding the `popstate` ramifications well enough, let's chat about it to make sure I'm not doing something stupid

Reply:
On second thought this check is unnecessary. We already check for duplicate entries during `popstate`: https://github.com/rerun-io/rerun/blob/andreas/dataplatform-url-update/crates/viewer/re_viewer/src/history.rs#L142

Reply:
yeah, if I don't do this we accumulate too many entries here - e.g. switching between the application id view & the recording view generates new entries that we can't actually distininguish.

All of these here except the last are the same page:
<img width="130" height="207" alt="image" src="https://github.com/user-attachments/assets/1f96ba6e-124e-4c86-9420-2e3a59598799" />
Guess we should look into giving those better names... later :)

Reply:
I believe changing the names of history entries is impossible. There's a `title` parameter in `replaceEntry` that no browser currently does anything with. MDN call it `unused`: https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState
Issue for context: [whatwg/html#2174](https://www.github.com/whatwg/html/issues/2174)